common/lib/common-utils/src/idleTaskScheduler.ts

```diff
@@ -0,0 +1,47 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryProperties } from "@fluidframework/common-definitions";
+
+/**
+ * API used to schedule non-essential tasks
+ * Time measurements are in milliseconds as a floating point with a decimal
+ */
+export class IdleTaskScheduler {
+  /*
+  Takes in and runs a callback during idle time. Fallback to setTimeout if window doesn't
+  support requestIdleCallback.
+  @returns A promise pertaining to the callback that was passed in.
+  */
+  public async scheduleIdleTask<T>(callback: () => T, timeout: number, props: ITelemetryProperties = {}): Promise<T> {
+    // let promise;
+    // Check for the availability in window.
+    if (typeof window?.requestIdleCallback === "function") {
+      return  new Promise<T>((resolve, reject) => {
+        requestIdleCallback(doLowPriorityTask, { timeout });
+        function doLowPriorityTask(deadline): void {
```

Comment:
While JS allows it, I find it odd to use symbol before it is defined.
Consider defining doLowPriorityTask you use it. Also, it's more common to do it that way:
```ts
const doLowPriorityTask = (deadline) => {
    try {
        resolve(callback());
    } catch (e) {
        reject(e);
    }
};
```